### PR TITLE
Closes #55 

### LINKED ISSUE #55: buildFolders and macroLocation updates
When building different targets (eg sas9 or viya) the available functionality is different.  Hence different sets of folders need to be built.  We need a way to only build certain folders for each target.

Definition of done:

The `buildFolders` attribute in `config.json` should be renamed `cmnServices` and the default value should be `["sas/services"]` instead of `["services"]`

Each `targets` array object should also have a `tgtServices` array attribute.  When building that target, all SAS files in the `cmnServices` and `tgtServices` files are created in the `sasbuild` folder.  The order should not matter.

The `macroLocations` attribute should be renamed to `cmnMacros` 

The `targets` attribute `macroSourcePath` should be changed to `tgtMacros` and be an array attribute.  The dependencies in these folders should be considered in the order they are encountered in the array, and then by the folders in the `cmnMacros` array.  The first one found is the 'winner'.

This will be a breaking change, so should be committed with the following syntax:

```
feat:  enabling target specific macro locations.
BREAKING CHANGE: the buildFolders and macroLocations attributes in config.json have changed to cmnServices and cmnMacros respectively
```

### PR DESCRIPTION
**Pull request by saadjutt01**

### CODE REVIEW

File: test/getDependencyPaths.spec.js [hunk 1]
```diff
@@ -108,7 +108,7 @@ describe("getDependencyPaths", () => {
     const result = prioritiseDependencyOverrides(
       dependencyNames,
       dependencyPaths,
-      "sas/sas9macros"
+      ["sas/sas9macros"]
```
Comment (allanbowe) [inline test/getDependencyPaths.spec.js:111]: does this need to be `sasjs` now?  (and below)

File: test/getDependencyPaths.spec.js [hunk 2]
```diff
@@ -144,7 +144,7 @@ describe("getDependencyPaths", () => {
...
```
Comment (allanbowe) [inline test/getDependencyPaths.spec.js:147]: and here

File: test/getDependencyPaths.spec.js [hunk 3]
```diff
@@ -126,7 +126,7 @@ describe("getDependencyPaths", () => {
...
```
Comment (allanbowe) [inline test/getDependencyPaths.spec.js:129]: here

File: test/getDependencyPaths.spec.js (hunk 1 above)
Comment (saadjutt01) [reply test/getDependencyPaths.spec.js]: as per our tgtMacros in config, it should be relative value and pass without parent folder

Comment (saadjutt01) [reply test/getDependencyPaths.spec.js]: updated to just `["sas9macros"]`

**Pull request merged**